app/lnd/lib/lightning.js

```diff
@@ -19,22 +19,64 @@ process.env.GRPC_SSL_CIPHER_SUITES =
     'ECDHE-ECDSA-CHACHA20-POLY1305'
   ].join(':')
 
-const lightning = (rpcpath, host) => {
+/**
+ * Creates an LND grpc client lightning service.
+ * @returns {rpc.lnrpc.Lightning}
+ */
+const lightning = async () => {
   const lndConfig = config.lnd()
-  const lndCert = fs.readFileSync(lndConfig.cert)
-  const sslCreds = grpc.credentials.createSsl(lndCert)
+
+  const host = lndConfig.lightningHost
   const rpc = grpc.load(lndConfig.lightningRpc)
 
-  const metadata = new grpc.Metadata()
-  const macaroonHex = fs.readFileSync(lndConfig.macaroon).toString('hex')
-  metadata.add('macaroon', macaroonHex)
+  const [sslCreds, macaroonCreds] = await Promise.all([
+    createSslCreds(lndConfig.cert),
+    createMacaroonCreds(lndConfig.macaroon)
+  ])
 
-  const macaroonCreds = grpc.credentials.createFromMetadataGenerator((params, callback) =>
-    callback(null, metadata)
-  )
   const credentials = grpc.credentials.combineChannelCredentials(sslCreds, macaroonCreds)
 
   return new rpc.lnrpc.Lightning(host, credentials)
 }
 
+/**
+ * Validates and creates the ssl channel credentials from the specified file path
+ * @param {String} certPath
+ * @returns {grpc.ChanelCredentials}
+ */
+const createSslCreds = async certPath => {
+  if (!(await validateFileAccess(certPath))) {
```

Comment:
According to https://nodejs.org/api/fs.html#fs_fs_access_path_mode_callback

> Using fs.access() to check for the accessibility of a file before calling fs.open(), fs.readFile() or fs.writeFile() is not recommended. Doing so introduces a race condition, since other processes may change the file's state between the two calls. Instead, user code should open/read/write the file directly and handle the error raised if the file is not accessible.

Reply:
Makes sense. I can rework this.